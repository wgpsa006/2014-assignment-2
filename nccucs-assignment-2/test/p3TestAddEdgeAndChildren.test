CreateGraph A
CreateGraph B
CreateNode n1 5
CreateNode n2 3
AddNode A n1
AddNode B n1
AddNode A n2
AddNode B n2
AddEdge A n1 n2
# Test that the only edge is from n1 to n2 in A
ListChildren A n1
ListChildren A n2
ListChildren B n1
AddEdge B n2 n1
# Test that the edge from n2 to n1 is added in B
ListChildren B n2
AddEdge A n1 n1
# Add an edge from n1 to itself in A and verify that B remains unchanged
ListChildren A n1
ListChildren B n1