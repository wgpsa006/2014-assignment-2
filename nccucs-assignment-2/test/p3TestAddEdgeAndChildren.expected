created graph A
created graph B
created node n1 with cost 5
created node n2 with cost 3
added node n1 to A
added node n1 to B
added node n2 to A
added node n2 to B
added edge from n1 to n2 in A
# Test that the only edge is from n1 to n2 in A
the children of n1 in A are: n2
the children of n2 in A are:
the children of n1 in B are:
added edge from n2 to n1 in B
# Test that the edge from n2 to n1 is added in B
the children of n2 in B are: n1
added edge from n1 to n1 in A
# Add an edge from n1 to itself in A and verify that B remains unchanged
the children of n1 in A are: n1 n2
the children of n1 in B are: